imagebuilder/builder/processManifest.go

```diff
@@ -188,8 +197,23 @@ func processManifest(manifestDir, rootDir string, bindMounts []string,
 	if err := cleanPackages(rootDir, buildLog); err != nil {
 		return err
 	}
-	if err := clearResolvConf(buildLog, rootDir); err != nil {
-		return err
+	// Check if resolv.conf has been modified by the image. If it has
+	// don't clear the file. If users have removed /etc/resolv.conf in
+	// the image for any reason don't blow up the build just eat the error
+	// and continue with the old logic of clearing resolv.conf.
+	var resolvStatEnd os.FileInfo
+	resolvEnd, err := os.Open(filepath.Join(rootDir, "/etc/resolv.conf"))
+	if err != nil {
+		resolvStatEnd = resolvStatStart
+	} else if resolvStatEnd, err = resolvEnd.Stat(); err != nil {
+		resolvStatEnd = resolvStatStart
+	}
+	defer resolvEnd.Close()
```

Comment:
The block of code above got a bit more messy than I would have liked and a tiny bit magical since in the case of any error reading resolv.conf we overwrite it with the original value to force the cleanup of the file.